### PR TITLE
Use grpc-netty-shaded instead of grpc-netty.

### PR DESCRIPTION
**Pull request by sduskis**
@igorbernstein2, the `google-cloud-java` BOM uses `grpc-netty-shaded`, so this would be a prereq for using the BOM.  I'm running into trouble with the `Provider` which is expecting this:

* com.google.bigtable.repackaged.io.grpc.netty.shaded.com.google.bigtable.repackaged.io.grpc.netty.NettyChannelProvider

Do you have any thoughts on how to fix the service file?

### CODE REVIEW

**Comment by sduskis** [comment]
This should work towards fixing #1756

**Comment by igorbernstein2** [comment]
Figured it out: the issue is caused by grpc-netty-shade.jar using the package name `io.grpc.netty.shaded.io.grpc.netty` and the  ServicesResourceTransformer usin greplaceAll:
https://github.com/apache/maven-shade-plugin/blob/ab82be37c5d4aa9855c03473a62cf7beed2f2b4a/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java#L219

To workaround the issue, please the following, right before the relocation of `io.grpc`:

```
                <!-- Take special care of grpc-netty-shaded, it uses the package
                     io.grpc.netty.shaded.io.grpc.netty, which will cause the
                     ServicesResourceTransformer to replace both occurrences of io.grpc
                 -->
                <relocation>
                  <pattern>io.grpc.netty.shaded.io.grpc.netty</pattern>
                  <shadedPattern>com.google.bigtable.repackaged.io.grpc.netty</shadedPattern>
                </relocation>
```

**Comment by sduskis** [comment]
@igorbernstein2, I update the pom.xml as per your suggestion, and now all tests pass!  Can you please review?

File: bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationAwaitCompletion.java [hunk 1]
```diff
@@ -26,12 +26,7 @@
 import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
+import java.util.concurrent.*;
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationAwaitCompletion.java]: nit: please revert the wildcard

**Pull request merged**